kotlintest-assertions/src/jvmMain/kotlin/io/kotlintest/properties/jvmgens.kt

```diff
@@ -153,3 +148,25 @@ inline fun <reified T : Enum<T>> Gen.Companion.enum(): Gen<T> = object : Gen<T>
 
 fun Gen.Companion.regex(regex: String) = RegexpGen(regex)
 fun Gen.Companion.regex(regex: Regex) = regex(regex.pattern)
+
+/**
+ * Returns a stream of values where each value is a randomly
+ * chosen File object from given directory. Directory does not exists, an empty sequence will be return.
+ * If recursive is true it gives files from inner directories as well recursively.
+ */
+fun Gen.Companion.file(directoryName: String, recursive: Boolean = false): Gen<File> = object : Gen<File> {
```

Comment:
I wonder if this should be `Gen.file` or `Gen.files`. @sksamuel ?

Reply:
I think either is fine.